ArmPkg/Drivers/ArmGic/GicV3/ArmGicV3Dxe.c

```diff
@@ -423,7 +423,7 @@ GicV3DxeInitialize (
       for (Index = 8; Index < (mGicNumInterrupts / 4); Index++) {
         MmioWrite32 (
           mGicDistributorBase + ARM_GIC_ICDIPTR + (Index * 4),
-          CpuTarget
+          (UINT32)CpuTarget

```

Comment:
This describes a 32-bit register. I would prefer to change CpuTarget to UINT32 over casting it on use.

Reply:
I do not disagree with using 32bit value. But the CpuTarget used for GICv3 is a 64bit value. Do you suggest we have another local variable for the legacy use case?

Reply:
Oh, I see. Thanks for clarifying.
Yes actually, I think it would be cleaner and clearer to define CpuTarget as a scope-local variable in each branch of

if (FeaturePcdGet (PcdArmGicV3WithV2Legacy)) {
} else [
}

I think it would be even cleaner to break all contents on each branch of the conditional completely out into helper functions. It's just confusing to handle such completely different cases in the same function.

But I'd be fine with just the scope-local definitions if that's your preference.

Reply:
Thanks for the suggestion. This is updated with the scope-local definitions.